crypto/zkp/primitives/commitments/damgard-fujisaki_square.go

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2017 XLAB d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package commitmentzkp
+
+import (
+	"math/big"
+
+	"fmt"
+
+	"github.com/xlab-si/emmy/crypto/commitments"
+)
+
+type DFCommitmentSquareProver struct {
+	equalityProver *DFCommitmentEqualityProver
```

Comment:
Maybe it would be better to embed `*DFCommitmentEqualityProver` (e.g. just remove `equalityProver`).

Reply:
Good point!